apps/hexagon_benchmarks/adb_run_on_device.sh

```diff
@@ -17,7 +17,7 @@ BIN=bin
 APP_TARGET=arm-64-android
 
 # Build the app.
-make bin/${APP_TARGET}/process
+make clean bin/${APP_TARGET}/process
```

Comment:
this seems like an antipattern. The main reason it would happen is if the makefiles don't have proper dependencies.

Reply:
@zvookin - I want to make sure that stale binaries are not pushed to the device. The dependencies in the Makefile are fine. We push the executable to the device *outside of the Makefile* in adb_run_on_device and that is why we make sure we clean before we build. There are 3 ways to ensure that stale binaries are not pushed to the device in the event that the compilation errored out - 1) Make sure that the rules for bin/%/process are such that the build always starts from scratch but that would be an inefficient use of Make 2) Always clean before make like in this patch. 3) look for the return value of make and if 0, only then push to the device. 

I see 2 and 3 as equivalent.

Reply:
The use case I am concerned about is using the script in active development -- e.g. trying to improve the schedule. Forcing a clean each time potentially increases the turnaround time significantly. (And it usually is totally unexpected by the developer. After things being slow for a long time they finally realize the script is at fault.) Not using the script is inconvenient as it encapsulates the arcane adb commands into a nice package.

A couple other approaches would be to add an argument to the script to force the clean or to break out the deployment part into another script that this one calls. (Or to just do a "make clean" before calling the script in the test harness where you need this behavior.)